yuniql-platforms/oracle/OracleDataService.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System.Collections.Generic;
+using System.Data;
+using Yuniql.Extensibility;
+using Oracle.ManagedDataAccess.Client;
+using System;
+using System.IO;
+using Yuniql.Extensibility.SqlBatchParser;
+using System.Linq;
+
+namespace Yuniql.Oracle
+{
+    ///<inheritdoc/>
+    public class OracleDataService : IDataService, IMixableTransaction
+    {
+        private string _connectionString;
+        private readonly ITraceService _traceService;
+
+        ///<inheritdoc/>
+        public OracleDataService(ITraceService traceService)
+        {
+            this._traceService = traceService;
+        }
+
+        ///<inheritdoc/>
+        public void Initialize(string connectionString)
+        {
+            this._connectionString = connectionString;
+        }
+
+        ///<inheritdoc/>
+        public bool IsTransactionalDdlSupported => false;
+
+        ///<inheritdoc/>
+        public bool IsSchemaSupported { get; } = true;
+
+        ///<inheritdoc/>
+        public bool IsBatchSqlSupported { get; } = true;
+
+        ///<inheritdoc/>
+        public bool IsUpsertSupported => false;
+
+        ///<inheritdoc/>
+        public string TableName { get; set; } = "__yuniql_schema_version";
+
+        ///<inheritdoc/>
+        public string SchemaName { get; set; }
+
+        ///<inheritdoc/>
+        public IDbConnection CreateConnection()
+        {
+            return new OracleConnection(_connectionString);
+        }
+
+        ///<inheritdoc/>
+        public IDbConnection CreateMasterConnection()
+        {
+            var masterConnectionStringBuilder = new OracleConnectionStringBuilder(_connectionString);
+            masterConnectionStringBuilder.DataSource = "INFORMATION_SCHEMA";
+
+            return new OracleConnection(masterConnectionStringBuilder.ConnectionString);
+        }
+
+        ///<inheritdoc/>
+        public List<string> BreakStatements(string sqlStatementRaw)
```

Comment:
If oracle does not support batch separator, you can just return entire sql statement such as
```csharp
            return new List<string> { sqlStatementRaw };
```

Reply:
There is support for BATCH statements in Oracle, if you can give clue about where its used. Eg: SELECT batching etc as ORACLE support MARS. DML statements can also be joined as we use that internally.

Reply:
The idea of batch here actually is simply grouping statements and executing them in a single connection.ExecuteNonQuery(). This is important for databases that does not support full transactional DDL such as MySql and Snowflake where each statement is an atomic operation. So to ensure yunql can wrap them in single transaction, we use bacth terminator such as "GO" in sql server.

Reply:
I dont think its realy to MARS as the DDL and DML scripts here do not return results.

Reply:
Ok, statement batching is supported in Oracle and we use ```;``` or ```/``` to batch statements. It seems like Oracle don't support Transactional DDL and we don't do that either. Doing ddl in Procedure etc also COMMIT the transaction as well. https://stackoverflow.com/questions/4711447/oracle-ddl-and-transaction-rollback

Reply:
In this case, it will behave like MySql and Snowflake. yuniql will pick all sql in the same file and attempt to execute them in single connection if possible. If it fails, the version will fail and the failing file and error infromation will be recorded in the table. Then any attempt to run will fail unless the issue is resolved manually.

To resolve the issue, user then have to manually fix the file and execute outside yuniql or split the file to another one and execute run with `--continute-after-failure` flag. The flag will skip the last failed sql file and move to next file in the last failing version.

All ideas welcome how to better handle this :)